duplicate_url_discarder/processors/subpath_removal.py

```diff
@@ -0,0 +1,37 @@
+import re
+from collections.abc import Iterable
+
+from scrapy.utils.url import parse_url
+
+from .base import UrlProcessorBase
+
+
+class SubpathRemovalProcessor(UrlProcessorBase):
+    """Removes the i-th subpath (starts at 0) from the URL path if it exists.
+
+    An example would be removing the 0-th and 2nd subpath of a URL like
+    https://example.com/a/b/c/d would result in https://example.com/b/c.
```

Comment:
From my understanding of the statement:

```suggestion
    https://example.com/a/b/c/d would result in https://example.com/b/d.
```

Reply:
Good catch!